Pull request template reworked (#4317)
TeamNewPipe/NewPipe: .github/PULL_REQUEST_TEMPLATE.md

<!-- Hey there. Thank you so much for improving NewPipe, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

> <!-- While bullet points are the norm in this section, feel free to write a text instead of a list -->

Reviewer: ```suggestion
<!-- While bullet points are the norm in this section, feel free to use sentences -->
```

> <!-- While bullet points are the norm in this section, feel free to write a text instead of a list -->

Author: ```suggestion
<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->
```

- record videos
- create clones
- take over the world

#### Fixes the following issue(s)

> <!-- Also add non-reddit or other links relevant to your change. -->

Reviewer 2: Why non-reddit? reddit is explicitly allowed.

Author: That would be Reddit. I mean "reddit" as in the verb, meaning questionable content.

Reviewer 2: Ah. ok :D We meant Reddit.

Author: Yes, but saying it like that would be entrapment. :)

Reviewer: > I mean "reddit" as in the verb, meaning questionable content.

I don't think 'reddit' is a word in English at all.

> <!-- Also add non-reddit or other links relevant to your change. -->

Author: ```suggestion
<!-- Also add any other links relevant to your change. -->
```

- 

#### Relies on the following changes
<!-- Delete this if it doesn't apply to you. -->
- 

#### APK testing 

> <!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe. (No names like "patch-0" and "feature-1".) "commentfix" is good if your PR implements a bugfix for comments. -->

Reviewer: ```suggestion
<!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe, e.g. "commentfix", if your PR implements a bugfix for comments. (No names like "patch-0" and "feature-1".)  -->
```

debug.zip

> #### Due dilligence

Reviewer: ```suggestion
#### Due diligence
```

> - [ ] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

Reviewer 2: Why did you remove the last part?

Author: There isn't any agreement needed. This is a scourge, and should be avoided.
